ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -455,6 +465,32 @@ private[kvutils] class TransactionCommitter(
           RejectionReason.Inconsistent("Causal monotonicity violated")))
   }
 
+  /**
+    * Validates that 'contract key' -> 'contract id' mapping resolved by a participant is still valid.
+    */
+  private def validateContractKeysConsistency(
+      currentResolvedIds: Map[DamlContractKey, String]
+  )(
+      recordTime: Option[Timestamp],
+      transactionEntry: DamlTransactionEntrySummary,
+  ): StepResult[DamlTransactionEntrySummary] = {
+    val keysAreConsistent: Boolean = transactionEntry.contractKeyToIdMap.forall {
+      case (contractKey, contractId) =>
+        // Given a mapping contractKey -> contractId we can treat it as valid if either of following conditions apply:
+        // 1. there is no contractKey in the current ledger state - treat contractKey -> contractId as valid
+        // 2. there is an entry for contractKey in the current ledger state - verify that the contractId matches the contract id in the ledger state
```

Comment:
What's the rationale for these rules? They are not obviously correct to me, especially the first item.

As this test determines the rejection reason, I recommend to be a bit more careful. Depending on your trust model, you may want to consider cases such as the following:

* If a buggy/malicious participant sends a contract key-contract id pair where the resolved contract does not fit to the given key (e.g., a different template or no key at all), then it would make more sense to `dispute` the submission rather than say it's `inconsistent`.
* If a buggy/malicious participant sends that a key should resolve to a contract that is created only as part of this transaction, then this is "obviously wrong", so `disputed` makes sense. (This only makes sense if we 
* The submitters do not see a contract key resolution during submission, but the submitters may see the contract itself through divulgence. That's expected behaviour and therefore should be `inconsistent`.